src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11613,8 +11620,10 @@ void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix
     } else {
       evex_v = (nds_enc >= 16);
     }
+    bool eevex_x = adr.index_needs_rex2() && adr.index()->encoding() >= 24;
+    bool evex_b = adr.base_needs_rex2() && ((adr.base()->encoding() & 16) == 16);
```

Comment:
Could this be:
    bool eevex_x = adr.index_needs_rex2();
    bool evex_b = adr.base_needs_rex2();

Reply:
Also has_gpr argument can be removed.